### PR TITLE
support dshuttle storage type

### PR DESCRIPTION
**Pull request by Binyang2014**

### CODE REVIEW

File: src/rest-server/src/models/v2/storage.js [hunk 1]
```diff
@@ -90,6 +90,12 @@ const convertVolumeDetail = async (pvc) => {
     if (pv.spec.flexVolume.options.mountoptions) {
       storage.mountOptions = pv.spec.flexVolume.options.mountoptions.split(',');
     }
+  } else if (pv.spec.csi) {
+    if (pv.spec.csi.driver === 'dshuttle') {
+      storage.type = 'dshuttle';
+      storage.data = {};
```
Comment (abuccts) [inline src/rest-server/src/models/v2/storage.js:96]: why the data is empty

File: src/rest-server/src/models/v2/storage.js [hunk 2]
```diff
@@ -90,6 +90,12 @@ const convertVolumeDetail = async (pvc) => {
     if (pv.spec.flexVolume.options.mountoptions) {
       storage.mountOptions = pv.spec.flexVolume.options.mountoptions.split(',');
     }
+  } else if (pv.spec.csi) {
+    if (pv.spec.csi.driver === 'dshuttle') {
+      storage.type = 'dshuttle';
```
Comment (abuccts) [inline src/rest-server/src/models/v2/storage.js:95]: update api docs accordingly

Comment (Binyang2014) [reply src/rest-server/src/models/v2/storage.js:95]: DShuttle just used by internal users in this release. Will update API when DShuttle published

File: src/rest-server/src/models/v2/storage.js (hunk 1 above)
Comment (Binyang2014) [reply src/rest-server/src/models/v2/storage.js]: Fixed

File: src/rest-server/src/models/v2/storage.js [hunk 3]
```diff
@@ -93,9 +93,12 @@ const convertVolumeDetail = async (pvc) => {
   } else if (pv.spec.csi) {
     if (pv.spec.csi.driver === 'dshuttle') {
       storage.type = 'dshuttle';
-      storage.data = {};
+      storage.data = pv.spec.csi.volumeAttributes;
     }
     storage.readOnly = pv.spec.csi.readOnly === true;
+    if (pv.spec.mountOptions) {
```
Comment (abuccts) [inline src/rest-server/src/models/v2/storage.js:99]: unnecessary condition

Comment (Binyang2014) [reply src/rest-server/src/models/v2/storage.js]: fixed

**Review by abuccts — APPROVED** [review]

**Pull request merged**